src/util/restoreSelection.js

```diff
@@ -10,50 +10,49 @@ import {
   isDivider,
   isRoot,
   pathToContext,
-  setSelection,
+  setSelection
 } from '../util.js'
 
 /** Restores the selection to a given editable thought and then dispatches setCursor. */
 // from the element's event handler. Opt-in for performance.
 // asyncFocus.enable() must be manually called before when trying to focus the selection on mobile
 // (manual call since restoreSelection is often called asynchronously itself, which is too late for asyncFocus to work)
 export const restoreSelection = (thoughtsRanked, { offset, cursorHistoryClear, done } = {}) => {
-
   // no selection
   if (!thoughtsRanked || isRoot(thoughtsRanked) || isDivider(headValue(thoughtsRanked))) return
 
   const thoughts = pathToContext(thoughtsRanked)
 
   // only re-apply the selection the first time
   if (!globals.disableOnFocus) {
-
     globals.disableOnFocus = true
 
-    // use current focusOffset if not provided as a parameter
-    const focusOffset = offset != null
-      ? offset
-      : window.getSelection().focusOffset
-
-    store.dispatch({ type: 'setCursor', thoughtsRanked, cursorHistoryClear })
-
-    // re-apply selection
     setTimeout(() => {
+      const offsetFromStorage = localStorage.getItem('currentCursorPosition')
+      // use current focusOffset if not provided as a parameter
+      const focusOffset = offset || (offsetFromStorage !== undefined ? Number(offsetFromStorage) : window.getSelection().focusOffset)
```

Comment:
I'm concerned about moving the `focusOffset` definition into the `setTimeout` rather than before the `setTimeout`, as there are some sensitive timing issues here. Can you explain what motivated this move? I will test it on mobile since the selection behaviors differs slightly on mobile, in particular the usage of `asyncFocus`.

Reply:
I'll reconsider this. There are too many setTimeouts here. Thanks!